packages/x-date-pickers/src/hooks/usePickersFieldContext.ts

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { PickersFieldContext } from '../internals/components/PickersFieldProvider';
+
+/**
+ * Returns the context passed by the picker that wraps the current field.
+ */
+export const usePickersFieldContext = () => {
```

Comment:
We will need to create a `usePickersFieldNullableContext` hook in `PickersFieldProvider.tsx` for our built in fields that can be used as standalone component.

Reply:
At first I added a param `<TOptional>(isOptional: TOptional = false` to this hook to allow people to configure the hook to throw an error or not when the context is null (and have the typing reflect that decision).

But I think 2 hooks is just simpler that the added complexity on the public API.